### ISSUE TITLE
Do not provide "origin" to plugins when invoking their registered RPC methods without permission

### ISSUE THREAD

**Issue by wighawag** [MetaMask/metamask-plugin-beta#30]
hey great initiative and great work with the plugin system
That's really cool to have that to experiment new things!

The idea I would like to implement is an encryption/ decryption api

For that I want to get access to the origin of the app requesting encryption/decryption (the app using the plugin)
This is so that encryption is sandboxed to the origin of the app, and not of the plugin (and that is why I don't want to use getAppKey only). This is basically an implementation of what I described here for safe automatic encryption/decryption : https://medium.com/@wighawag/3-proposals-for-making-web3-a-better-experience-974f97765700

For Metamask to preserve privacy though, I should not be allowed to use `fetch` as this would allow me to track what apps / website the users of the plugin are using.

In other words, access to `fetch` should be a permission.

Few added thoughts: 

Maybe the 'fetch' permission can be granted by default if no other privacy-breaching permission is given.
But I think the simplest is to simply make `fetch` always an explicit permission

Note that there is an alternative solution to get an encryption key per app's origin without revealing the origin to the plugin:
It would be to have another wallet api : `wallet.getAppKeyForCurrentBrowsingContext`
But I would argue that is still privacy-breaching as it would allow the plugin to track how many time the user is using that context.

**Comment by danfinlay** [comment]
Fun find: at 23 minutes into this talk from 2002, Mark Miller suggests the same thing as you (add a special warning if gaining both disk and network access):
https://youtu.be/KoM_aCuFk1w

**Comment by wighawag** [comment]
Cool :)
It is crazy how much we need to re-learn/re-discover, no matter how much was already done.
I am really happy to know you are digging all that info :)
